android/KMEA/app/src/main/java/com/tavultesoft/kmea/ModelPickerActivity.java

```diff
@@ -155,9 +156,19 @@ public void onItemClick(AdapterView<?> parent, View view, final int position, lo
             // While awkward, we must obtain the preInstalledModelMap before any installations occur.
             // We don't want to remove the model we just installed, after all!
             if(preInstalledModelMap != null) {
-              LexicalModel preInstalled = new LexicalModel(preInstalledModelMap);
+              // This might be unncessary
+              LexicalModel preInstalled = new LexicalModel(
+                preInstalledModelMap.get(KMManager.KMKey_PackageID),
+                preInstalledModelMap.get(KMManager.KMKey_LexicalModelID),
+                preInstalledModelMap.get(KMManager.KMKey_LexicalModelName),
+                preInstalledModelMap.get(KMManager.KMKey_LanguageID),
+                preInstalledModelMap.get(KMManager.KMKey_LanguageName),
+                preInstalledModelMap.get(KMManager.KMKey_LexicalModelVersion),
+                preInstalledModelMap.get(KMManager.KMKey_HelpLink),
+                "" // String modelURL
+                );
```

Comment:
This is a step backwards. But will get fixed when `KMManager.getAssociatedLexicalModel()` is updated to return LexicalModel instead of HashMap (in a separate PR)